src/elements/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+import List, { ListProps } from "@vertigis/web/ui/List";
+import ListItemIcon from "@vertigis/web/ui/ListItemIcon";
+import ListItemText from "@vertigis/web/ui/ListItemText";
+import ListItemButton from "@vertigis/web/ui/ListItemButton";
+import { UIContext } from "@vertigis/web/ui/UIContext";
+import { BrandingService } from "@vertigis/web/branding/BrandingService";
+/**
+ * Properties for the `Tree` component.
+ */
+export declare type TreeViewProps = {
+    showCheckBoxes?: boolean;
+};
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+
+type SettableTreeProps = Pick<ListProps, "dense" | "subheader">;
+
+type SettableCheckBoxProps = Pick<TreeViewProps, "showCheckBoxes">;
+
+interface TreeElementNode {
+    key: string;
+    open: boolean;
+    primary: string;
+    secondary?: string;
+    link?: string;
+    icon?: DynamicIconProps["src"];
+    checked?: boolean;
+    children?: TreeElementNode[];
+}
+interface TreeElementProps
+    extends FormElementProps<TreeElementNode[] | undefined>,
+        SettableBoxProps,
+        SettableTreeProps,
+        SettableCheckBoxProps {
+    items: TreeElementNode[];
+    onClick?: (node: TreeElementNode) => void;
+    onMouseEnter?: (node: TreeElementNode) => void;
+    onMouseLeave?: (node: TreeElementNode) => void;
+}
+
+function flattenNodes(node: TreeElementNode, collection: TreeElementNode[]) {
+    if (node.checked) {
+        collection.push(node);
+    }
+    if (node.children && node.children.length > 0) {
+        for (const child of node.children) {
+            flattenNodes(child, collection);
+        }
+    } else {
+        return;
+    }
+}
+
+function filterTree(nodes: TreeElementNode[]) {
+    const children: TreeElementNode[] = [];
+    for (const node of nodes) {
+        flattenNodes(node, children);
+    }
+    return children;
+}
+
+function renderChild(
+    node: TreeElementNode,
+    index: number,
+    value: TreeElementNode[] | undefined,
+    showCheckBoxes: boolean | undefined,
+    level: number,
+    brandingService: BrandingService,
+    handleMouseEnter: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleMouseLeave: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleItemClick: (node: any) => void,
+    handleFolderClick: (node: any) => void,
+    handleSecondaryActionClick: (event: React.MouseEvent, node: TreeElementNode) => void
+) {
+    const isDense = brandingService.density === "compact";
+    const isFolder = Array.isArray(node.children);
+    const isNode = !node.children;
+    const m = 1.5 * level;
+    const icon = (
+        <ChevronDownIcon
+            fontSize="small"
+            className="gcx-layer-tree-node-expand-button"
```

Comment:
I don't think we can use any of these `gcx-layer-tree...` classes. We do want it to look like the layer list as much as possible, but we should not be relying on specific classes that it contributes.

Reply:
I've removed them - they didn't really impact the layout as much as I had expected.